build.gradle

```diff
@@ -2026,52 +2043,53 @@ project(':connect:runtime') {
   archivesBaseName = "connect-runtime"
 
   dependencies {
-
-    compile project(':connect:api')
-    compile project(':clients')
-    compile project(':tools')
-    compile project(':connect:json')
-    compile project(':connect:transforms')
-
-    compile libs.slf4jApi
-    compile libs.jacksonJaxrsJsonProvider
-    compile libs.jerseyContainerServlet
-    compile libs.jerseyHk2
-    compile libs.jaxbApi // Jersey dependency that was available in the JDK before Java 9
-    compile libs.activation // Jersey dependency that was available in the JDK before Java 9
-    compile libs.jettyServer
-    compile libs.jettyServlet
-    compile libs.jettyServlets
-    compile libs.jettyClient
-    compile(libs.reflections)
-    compile(libs.mavenArtifact)
-
-    testCompile project(':clients').sourceSets.test.output
-    testCompile libs.easymock
-    testCompile libs.junitJupiterApi
-    testCompile libs.junitVintageEngine
-    testCompile libs.powermockJunit4
-    testCompile libs.powermockEasymock
-    testCompile libs.mockitoCore
-    testCompile libs.httpclient
-
-    testCompile project(':clients').sourceSets.test.output
-    testCompile project(':core')
-    testCompile project(':core').sourceSets.test.output
-
-    testRuntime libs.slf4jlog4j
+    implementation project(':connect:api')
```

Comment:
Thoughts @kkonstantine and @rhauch? I thought connectors were meant to reference `connect-api` for compilation and the runtime is what gets included at... runtime. :)

Reply:
ya, I agree that referencing connect-runtime in compile time is weird. However, core module is allowed to propagate client module. It would be better to get consistent rule for both connect-runtime and core.

Reply:
Connector projects do require the `connect-api` module as a compile-time dependency, but most (if not all) will include `connect-runtime` as a test dependency so that they can run their connector in Connect runtime.

I don't really see anything that changed relative to the compile-time dependencies. In particular, `connect-api` was included before and is again, but the blank line (2029) was removed, causing this diff page to render a bit strange.

Reply:
The difference is that `connect-runtime` doesn't expose `connect-api` and others transitively anymore. If `connect-runtime` is used for tests where people _only_ depend on `connect-runtime` (this is an anti-pattern, but easy to use), things may break for them. That is the reason why I used `api` for the `clients` dependency in `core`, so I'll do the same for `connect-runtime`.

Reply:
Ah, I see that now. If we wanted to be strict, we could leave as an implementation dependency. However, I agree that it's safest to use `api` for the `connect-api` module.

Reply:
Fixed.

Reply:
@rhauch Can you take a look at the connect build changes and check if it's all good?

Reply:
The Connect portion looks good, so approving on this aspect. Thanks!